client/Lykke.AlgoStore.MatchingEngineAdapter.Client/RequestManager.cs

```diff
@@ -59,13 +58,10 @@ private void SendAuthRequest()
 
             task.ContinueWith((t) =>
             {
-                t.Result.ContinueWith((inner) =>
-                {
-                    var response = inner.Result.Message as PingRequest;
+                var response = t.Result.Message as PingRequest;
```

Comment:
I see two opportunities here
-  Make this method async|await and calls from a lambda on top, where you subscribing to an event
-  It is not clear who should the UnauthorizedAccessException? the OnConnectionEstablished? It is not a best practice throwing exceptions from event handles.

Reply:
The same question about exception for ProcessResponse